openassessment/data.py

```diff
@@ -993,15 +999,37 @@ def create_zip_with_attachments(cls, file, course_id, submission_files_data):
 
         with ZipFile(file, 'w') as zip_file:
             for file_data in submission_files_data:
-                file_content = (
-                    cls._download_file_by_key(file_data['key'])
-                    if file_data['type'] == cls.ATTACHMENT
-                    else file_data['content']
-                )
-
-                zip_file.writestr(file_data['file_path'], file_content)
-                csvwriter.writerow(file_data)
+                clone_file_data = copy.copy(file_data)
+                key = clone_file_data['key']
+                file_path = clone_file_data['file_path']
+                try:
+                    file_content = (
+                        cls._download_file_by_key(key)
+                        if clone_file_data['type'] == cls.ATTACHMENT
+                        else clone_file_data['content']
+                    )
+                except FileMissingException:
+                    clone_file_data['file_found'] = False
+                    # added a header to csv file to indicate that the file was found or not.
+                    # not sure if I should create a {file_path}.error.txt to indicate the file error more clearly.
+                    logger.warning(
+                        'File: \'{file_path}\' with key: \'{key}\' was missing.\n'
```

Comment:
Make log more explicit. What does "missing" mean? To make searching easier, propose making the first part a static string and logging out info afterwards. Proposed:

```suggestion
                        'File for submission could not be downloaded for ORA submission archive.\n'
```

Also include updated log (so we can see what those fields look like) in the PR description for verification. You may also find that some info is redundant and can be removed from the log.